app.js

```diff
@@ -0,0 +1,101 @@
+    history: 'She is a magazine editor and author and she will publish all activities that will be held at the event in her magazine',
+  },
+  {
+    id: 5,
+    pic: './images/lila.png',
+    altText: 'featured speaker image',
+    nameOfSpeaker: 'Stephanie',
+    designation: 'Southafrican Air news Magazine Author and Editor with 20years experience',
+    history: 'She is a magazine editor and author and she will publish all activities that will be held at the event in her magazine.',
+  },
+  {
+    id: 6,
+    pic: './images/reagan.png',
+    altText: 'featured speaker image',
+    nameOfSpeaker: 'Reagan Storch',
+    designation: 'Zambian born experienced photographer and Graphic Designer with 15 years experience',
+    history: 'He is a photographer, graphic designer, Burning cirle airshow sponsor and the founder of International studios.',
+  },
+];
+
+const speakerContainer = document.createElement('section');
+speakerContainer.setAttribute('class', 'featured-speakers');
+speakerContainer.setAttribute('id', 'speakers');
+
+data.forEach((speakerPara) => {
+  const speakerSection = document.createElement('div');
+  speakerSection.setAttribute('class', 'speaker-section');
+  speakerSection.setAttribute('id', 'each-speaker');
+
+  const imageSection = document.createElement('div');
+  imageSection.setAttribute('class', 'speaker-image-section');
+  const image = document.createElement('img');
+  image.setAttribute('class', 'speaker-image');
+  image.setAttribute('src', speakerPara.pic);
+  image.setAttribute('alt', speakerPara.altText);
+
+  imageSection.appendChild(image);
+  speakerSection.appendChild(imageSection);
+
+  const contentSection = document.createElement('div');
+  contentSection.setAttribute('class', 'speaker-details');
+  contentSection.setAttribute('id', 'about-speaker');
+
+  const speakerName = document.createElement('h2');
+  speakerName.setAttribute('class', 'speaker-name');
+  speakerName.textContent = speakerPara.nameOfSpeaker;
+  contentSection.appendChild(speakerName);
+
+  const speakerDesignation = document.createElement('p');
+  speakerDesignation.setAttribute('class', 'speaker-designation');
+  speakerDesignation.textContent = speakerPara.designation;
+  contentSection.appendChild(speakerDesignation);
+
+  const dotText = document.createElement('p');
+  dotText.setAttribute('id', 'dots');
+  dotText.textContent = '.........';
+  contentSection.appendChild(dotText);
+
+  const speakerHistory = document.createElement('p');
+  speakerHistory.setAttribute('class', 'speaker-history');
+  speakerHistory.textContent = speakerPara.history;
+  contentSection.appendChild(speakerHistory);
+
+  speakerSection.appendChild(contentSection);
+  speakerContainer.appendChild(speakerSection);
+});
+
+const body = document.querySelector('body');
+const partners = document.getElementById('partners');
+body.insertBefore(speakerContainer, partners);
```

Comment:
- [ ] The iteration over articles in `"Featured speakers"` section should be done using the `window.onload` function, please update your `js` files (the onload function makes sure that the minimum required assets have been loaded successfully).
- [ ] I suggest that you use either `appendChild()` or `innerHTML()` instead of `insertBefore` to dynamically attach the content.

Reply:
And if i may ask what is so wrong with how i did the iteration for the featured speakers. I mean thats your method not a microverse method the fact that it works fine shouldnt be a problem.

Reply:
Hi @KayLemba,
my apologies again, `insertBefore` is not used in the correct context in this case since it does not append the `partners` into the correct Featured Speakers `section` it instead appends it to the `body` tag. again I apologize if this was a bit unclear, I should have explained more why this change was required.

Reply:
done